microdata_validator/metadata_model.py

```diff
@@ -0,0 +1,349 @@
+        self.label = variable_dict["label"]
+        self.data_type = variable_dict["dataType"]
+        self.variable_role = variable_dict["variableRole"]
+        self.format = variable_dict.get("format", None)
+        self.key_type = (
+            None if "keyType" not in variable_dict
+            else KeyType(variable_dict["keyType"])
+        )
+        self.represented_variables = [
+            RepresentedVariable(represented_variable_dict)
+            for represented_variable_dict
+            in variable_dict["representedVariables"]
+        ]
+
+    def get_key_type(self):
+        return None if self.key_type is None else self.key_type.name
+
+    def to_dict(self) -> dict:
+        dict_representation = {
+            "name": self.name,
+            "label": self.label,
+            "dataType": self.data_type,
+            "variableRole": self.variable_role,
+            "representedVariables": [
+                represented_variable.to_dict()
+                for represented_variable in self.represented_variables
+            ]
+        }
+        if self.format is not None:
+            dict_representation["format"] = self.format
+        if self.key_type is not None:
+            dict_representation["keyType"] = self.key_type.to_dict()
+        return dict_representation
+
+    def patch(self, other: 'Variable'):
+        if other is None:
+            raise PatchingError(
+                'Can not delete Variable'
+            )
+        if (
+            self.name != other.name or
+            self.data_type != other.data_type or
+            self.format != other.format or
+            self.variable_role != other.variable_role
+        ):
+            raise PatchingError(
+                'Illegal change to one of these variable fields: '
+                '[name, dataType, format, variableRole]]'
+            )
+        self.label == other.label
+        if self.key_type is None and other.key_type is not None:
+            raise PatchingError(
+                'Can not change keyType'
+            )
+        if self.key_type is not None:
+            self.key_type.patch(other.key_type)
+        if len(self.represented_variables) != len(other.represented_variables):
+            raise PatchingError(
+                'Can not add or delete represented variables.'
+            )
+        for idx in range(len(self.represented_variables)):
+            self.represented_variables[idx].patch(
+                other.represented_variables[idx]
+            )
+
+
+class IdentifierVariable(Variable):
+    pass
+
+
+class MeasureVariable(Variable):
+    pass
+
+
+class AttributeVariable(Variable):
+    pass
+
+
+class Metadata:
+    name: str
+    temporality: str
+    language_code: str
+    population_description: str
+    subject_fields: List[str]
+    temporal_coverage: TimePeriod
+    measure_variable: MeasureVariable
+    identifier_variable: IdentifierVariable
+    start_variable: AttributeVariable
```

Comment:
Are you sure that every dataset in the system has only 2 attribute variables?

Reply:
Looked around and there are some datasets in the first version that has extra attributes actually, so this does need an update 👍🏻 Seems like it's never done any more (Maybe it was decided to be unecessary?), but this has to be backwards compatible either way so I will fix!